Naloga3/src/Naloga3.jl

```diff
@@ -0,0 +1,233 @@
+
+Sistem:
+    dy1/dt = y2
+    dy2/dt = -(g / l) * sin(y1)
+"""
+function f(t, y, g, l)
+    return [y[2], -(g / l) * sin(y[1])]
+end
+
+
+"""
+    DOPRI(f, y0, tspan, h, g, l)
+
+Resi sistem y' = f(t,y) z metodo Dormand–Prince 5. reda (DOPRI5).
+Implementacija uporablja Butcherjevo tabelo in genericno zanko.
+
+Vhodni podatki:
+- `f`: funkcija desne strani (mora biti f(t, y, g, l))
+- `y0`: zacetni pogoj (vektor dolzine 2)
+- `tspan`: casovni interval kot par (t0, t1)
+- `h`: dolzina koraka (npr. 0.01)
+- `g`, `l`: parametra modela
+
+Vrne:
+- seznam casov `ts`
+- seznam resitev `ys` (vsaka je vektor dolzine 2)
+"""
+function DOPRI(f, y0, tspan, h, g, l)
+    # koeficienti DormandPrinc, Butcherjeva tabela
+    c = [0,
+         1//5,
+         3//10,
+         4//5,
+         8//9,
+         1]
+
+    a = [
+        [],                                         
+        [1//5],                                     
+        [3//40,         9//40],                     
+        [44//45,      -56//15,       32//9],        
+        [19372//6561, -25360//2187, 64448//6561, -212//729],   
+        [9017//3168,  -355//33,     46732//5247,   49//176, -5103//18656]  
+    ]
+
+    b = [35//384, 0, 500//1113, 125//192, -2187//6784, 11//84]  # uteži za 5. red
+
+    # inicializacija trenutni čas in začetno stanje
+    t0, t1 = tspan
+    t = t0
+    y = y0
+
+    ts = [t]
+    ys = [y0]
+
+    while t < t1
+        #če presegamo konec intervala, prilagodimo korak
+        if t + h > t1
+            h = t1 - t
+        end
+
+        # shrani vektorje k
+        ks = Vector{Vector{Float64}}(undef, 6)
+
+        #vmesna stanja izračun
+        for i in 1:6
+            # sestej prispevke prejsnjih k
+            yi = y #zacetna vrednost
+            if !isempty(a[i]) #ni prvi korak
+                s = zero(y)
+                for j in 1:length(a[i])
+                    s += a[i][j] * ks[j]
+                end
+                yi = y + h * s
+            end
+            ti = t + c[i]*h
+            ks[i] = f(ti, yi, g, l)
+        end
+
+        # posodobitev rešitve z uteženim povprečjem vmesnih korakov
+        s = zero(y)
+        for i in 1:6
+            s += b[i] * ks[i]
+        end
+        y = y + h * s
+        t += h
+
+        push!(ts, t)
+        push!(ys, y)
+    end
+
+    return ts, ys
+end
+
+
+"""
+    poisci_periodo(ts, ys)
+
+Preprosto oceni periodni cas nihala iz rešitve.
+Najde dva zaporedna lokalna maksimuma (vrhova theta) in izracuna razliko casov.
+
+Vrne ocenjen periodni cas (ali NaN, ce ne najde).
+"""
+function poisci_periodo(ts, ys)
+    thetas = [y[1] for y in ys] #koti
+    max_indeksi = []
+
+    for i in 2:length(thetas)-1
+        if thetas[i-1] < thetas[i] && thetas[i+1] < thetas[i]
+            push!(max_indeksi, i)
+        end
+    end
+
+    if length(max_indeksi) >= 2
+        i1, i2 = max_indeksi[1], max_indeksi[2]
+        return ts[i2] - ts[i1]  # celotna perioda
```

Comment:
Natančnost izračun periode je omejena s korakom metode Dopri5. Glede na to, da je korak izbran adaptivno, nimate nobene kontrole, kako natančen bo rezultat. 

Problem iskanja periode lahko formulirate kot reševanje nelinearne enačbe $y(t+p) = y(t)$. In nato uporabite eno od metod za reševanje nelinearnih enačb (bisekcija, Newtonova metoda, ...).